packages/core/src/awsService/cloudWatchLogs/commands/tailLogGroup.ts

```diff
@@ -147,6 +163,48 @@ function trimOldestLines(
     edit.delete(document.uri, range)
 }
 
+function updateStatusBarItemsOnStreamEvent(session: LiveTailSession, event: LiveTailSessionUpdate) {
+    updateIsSampled(session, event)
+    updateEventRate(session, event)
+}
+
+function updateIsSampled(session: LiveTailSession, event: LiveTailSessionUpdate) {
+    let isSampled: boolean
+    //sessionMetadata is expected to always be present on a LiveTail stream update
+    if (event.sessionMetadata === undefined || event.sessionMetadata.sampled === undefined) {
+        isSampled = false
+    } else {
+        isSampled = event.sessionMetadata.sampled
+    }
+    session.isSampled = isSampled
+}
+
+function updateEventRate(session: LiveTailSession, event: LiveTailSessionUpdate) {
+    let numEvents
+    if (event.sessionResults === undefined) {
+        numEvents = 0
+    } else {
+        numEvents = event.sessionResults.length
+    }
+    session.eventRate = numEvents
+}
+
+function hideShowStatusBarItemsOnActiveEditor(session: LiveTailSession, document: vscode.TextDocument) {
+    vscode.window.onDidChangeActiveTextEditor((editor) => {
+        if (editor?.document === document) {
+            session.showStatusBarItem()
+        } else {
+            session.hideStatusBarItem()
```

Comment:
There is never a need for multiple "enable/disable" functions. Just make a parameter.


```suggestion
            session.showStatusBarItem(false)
```